cube/src/main/java/zone/cogni/asquare/cube/sort/StringRdfVisitor.java

```diff
@@ -49,6 +52,40 @@ public String visitLiteral(Literal literal) {
     throw new RuntimeException("how can this be possible? literal " + literal);
   }
 
+  @Override
+  public String visitStmt(Resource r, Statement statement) {
+    String result = "";
+
+    Resource subject = statement.getSubject();
+    Property predicate = statement.getPredicate();
+    RDFNode object = statement.getObject();
+
+    // visit subject
+    if (subject.isAnon()) {
```

Comment:
you do not need this. the idea of a visitor is to do something like

`String result = (String) rdfNode.visitWith(StringRdfVisitor.instance);`

so visitor takes care of all types automagically.